src/libraries/System.Net.Quic/src/System.Net.Quic.csproj

```diff
@@ -128,7 +128,8 @@
   </ItemGroup>
 
   <!-- Support for deploying msquic -->
-  <ItemGroup Condition="'$(TargetsWindows)' == 'true' and
+  <!-- Native msquic assemblies should only be binplaced while running on windows. -->
+  <ItemGroup Condition="'$(HostOS)' == 'windows' and
```

Comment:
This is a bit of conflation.  Nothing about building on a Windows machine means that these files are present.  Targeting is the right concept.  Can we instead just exclude this from source build?  Since presumably the reason this isn't available is because we aren't building as part of source build.

Reply:
I'm actually surprised that the PackageReference worked but this didn't.  What failed that led you to add this?

Reply:
> I'm actually surprised that the PackageReference worked but this didn't. What failed that led you to add this?

sourcebuild leg failed.

Reply:
Of course something failed, how did it fail and why?

Reply:
> Can we instead just exclude this from source build? 

SourceBuild  leg if run on windows will not contain this file in the targeting pack.

> Targeting is the right concept.

why is the targeting the right concept here, we are binplacing the file in the runtime folders, shouldnt we check the runtime we are running on before binplacing ?

Reply:
> Of course something failed, how did it fail and why?

the missing fileVersion error that we started this journey with :P

Reply:
>  shouldnt we check the runtime we are running on before binplacing ?

The runtime we're running on has nothing to do with the product we are building, though it is sometimes related.  We could technically build all of libraries for Linux from a windows machine (and we used to do so at the start of .NETCore).

BinPlacing is supposed to involve a calculation on "best applicable configuration".  Typically we only binplace when the configuration being built is the best for a given target. https://github.com/dotnet/arcade/blob/22d90b34fe210ca9f85ea1896eda29f2de55f826/src/Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk/src/build/BinPlace.targets#L122

It looks like this project is bypassing that by directly defining BinPlaceDir items. In that case, this condition needs to do the work that would have been done by ChooseBestTargetFrameworksTask: the concept shouldn't be "host OS" but should be "target runtime for the vertical".  It looks like this should be `TargetOS`.

I'm still dubious about MSQuic packagereference though.  @MichaelSimons does this package even get built in source build repos?  Shouldn't we exclude the PackageReference?

Reply:
>I'm still dubious about MSQuic packagereference though. @MichaelSimons does this package even get built in source build repos? Shouldn't we exclude the PackageReference?

It does not get produced by source-build today.  Additionally `system.net.msquic.transport` is showing up as a prebuilt.  Excluding the PackageReference seems like the good approach for source-build.

Reply:
> The runtime we're running on has nothing to do with the product we are building, though it is sometimes related. We could technically build all of libraries for Linux from a windows machine 

i thought this applied to managed assemblies, i thought we cant build the native assemblies cross platform ?